### PR TITLE
Use channels to sync requests to mirrorlist repos

### PR DESCRIPTION
**Pull request by gw000**
Hi again, i hope i'm not making too many pull requests. Although i think #70 does solve the mirrorlist concurrency bugs, i still thought it was kind of ugly. This is an alternate implementation using channels instead of mutexes which removes all explicit locking. It also simplifies `pacoloco.go` by moving all the mirrorlist updates and repo case analysis (`URL` / `URLs` / `Mirrorlist`) behind the channel you ask for URL lists on.  I think it's a better solution and i'll close #70 if you agree.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -28,19 +32,29 @@ type RefreshPeriod struct {
...
+	UserAgent       string           `yaml:"user_agent"`
+	LogTimestamp    bool             `yaml:"set_timestamp_to_logs"`
 }
 
 var config *Config
 
+func initURLsChannel(name string, repo *Repo) {
+	in := make(chan chan []string)
+	go func() {
+		for out := range in {
```
Comment (anatol) [inline config.go:51]: Is it needed to be stopped at some point? Or it is ok to keep it running forever?

Comment (gw000) [reply config.go:51]: I think the only time you'd want to stop it is if you won't be handling any more URLs requests, so on exit in other words. It blocks until it gets a message over the channel, so it's not busy looping in the background or anything.

**Comment by anatol** [comment]
cc @Focshole for the review

**Comment by Focshole** [comment]
It took me a bit of time to review the code since i didn't knew go channel constructs. It looks perfect to me! No more concurrency issues in a safe and easy to read manner!

**Comment by Focshole** [comment]
I'll adapt my PR onto this work

**Comment by anatol** [comment]
Thank you for your feedback @Focshole. I will move ahead and merge this change.

**Comment by anatol** [comment]
@gw000 one more request. Could you please add more information to the commit description? Currently, the commit message is only `use channels to sync requests to mirrorlist repos`, but having more explanation of why the change was done will be useful in the future.

**Comment by gw000** [comment]
@anatol I added some details to the commit message, let me know if anything is missing.

@Focshole Sorry to make you review twice! I hadn't used channels before either. Thank you for this learning opportunity :smile:

**Comment by anatol** [comment]
Awesome, thank you for your work @gw000 !

**Pull request merged**